common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -341,6 +342,21 @@ public boolean contains(final UTF8String substring) {
     return false;
   }
 
+  public boolean contains(final UTF8String substring, int collationId) throws SparkException {
+    if (CollationFactory.fetchCollation(collationId).isBinaryCollation) {
+      return this.containsBinary(substring);
+    }
+    if (collationId == CollationFactory.LOWERCASE_COLLATION_ID) {
+      return this.toLowerCase().containsBinary(substring.toLowerCase());
+    }
+    // TODO: enable ICU collation support for "contains"
+    Map<String, String> params = new HashMap<>();
+    params.put("functionName", Thread.currentThread().getStackTrace()[1].getMethodName());
+    params.put("collationName", CollationFactory.fetchCollation(collationId).collationName);
+    throw new SparkException("COLLATION_NOT_SUPPORTED_FOR_FUNCTION",
```

Comment:
is this temporary?

Reply:
yes, will implement "contains" for non-binary collations in a future PR

Reply:
then we can throw an internal error for now, without an error class.

Reply:
I think the logic here was that there are gonna be some functions that are not (yet) supported for some particular collation types (some might get added at one point in the near future, while others may not), so we will eventually need a way to let users know. that's why we added the `COLLATION_NOT_SUPPORTED_FOR_FUNCTION` error class - while it will likely get removed from here, there are other places where it will appear instead